crates/iota-sdk-graphql-client/src/lib.rs

```diff
@@ -1477,12 +1478,43 @@ impl Client {
             let bcs = base64ct::Base64::decode_vec(result.effects.bcs.0.as_str())?;
             let effects: TransactionEffects = bcs::from_bytes(&bcs)?;
 
-            Ok(Some(effects))
+            if wait_for_finalization {
```

Comment:
I don't understand this, we're waiting for effects but still returning the effects we tried to parse before?